### PR TITLE
53 backend make send message class

### PR DESCRIPTION
**Pull request by ugedal1**
## Description
- Messages are now made with a GameMessageFactory
- All outgoing messages are defined in outgoing
- Websockethandler refactored to be more modular
- To make a new handler you need to implement the handler interface.
- Tests remade to use new systems
Fixes # (issue)

### Type of change

Please delete options that are not relevant.

- [x] Bug fix 
- [x] New functionality 
- [x] Design change
- [x] New tests
- [ ] Frontend change
- [x] Backend change

## How Has This Been Tested?
New tests have been made and all green
Please describe the tests that you ran / made to verify your changes. 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] The solution is responsive (remove if not relevant)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added tests for edge cases
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: interloperServer/src/main/java/com/interloperServer/interloperServer/websocket/GameWebSocketHandler.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Override
+    public void afterConnectionEstablished(@NonNull WebSocketSession session) {
+        System.out.println("WebSocket connected: " + session.getId());
+    }
+
+    @Override
+    public void afterConnectionClosed(@NonNull WebSocketSession session, @NonNull CloseStatus status) {
+        // keep as-is
```
Comment (supergelf) [inline interloperServer/src/main/java/com/interloperServer/interloperServer/websocket/GameWebSocketHandler.java:35]: ???

Comment (supergelf) [reply interloperServer/src/main/java/com/interloperServer/interloperServer/websocket/GameWebSocketHandler.java:35]: skal ikke dette være der lenger? /**
     * Removes player from lobby and game when they disconnect
     */
    @Override
    public void afterConnectionClosed(@NonNull WebSocketSession session, @NonNull CloseStatus status) {
        lobbyService.removeUser(session);

        // Check if the user was in a game
        for (String lobbyCode : gameManagerService.getAllGameCodes()) {
            gameService.handlePlayerDisconnect(session, lobbyCode);
        }

    }

**Review by supergelf — APPROVED** [review]
Lets get this merge!
